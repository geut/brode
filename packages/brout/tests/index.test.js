import { Brout } from '../src/index.js'
import tapParser from '../src/parsers/tap.js'
import uvuParser from '../src/parsers/uvu.js'

const logger = logs => ({ type, args }) => {
  logs.push({ type, content: args.join(' ') })
}

test('basic', async () => {
  const logs = []

  const brout = new Brout({
    url: 'http://127.0.0.1:3000',
    command: 'webpack serve ./tests/fixtures/basic.js --config ./tests/webpack.config.js',
    log: logger(logs)
  })

  await expect(brout.run()).resolves.toBe(0)
  expect(logs).toMatchSnapshot()
})

test('basic fail', async () => {
  const logs = []

  const brout = new Brout({
    url: 'http://127.0.0.1:3000',
    command: 'webpack serve ./tests/fixtures/basic-fail.js --config ./tests/webpack.config.js',
    log: logger(logs)
  })

  await expect(brout.run()).rejects.toThrow('error0')
  expect(logs).toMatchSnapshot()
})

test('tap', async () => {
  const logs = []

  const brout = new Brout({
    url: 'http://127.0.0.1:3000',
    command: 'webpack serve ./tests/fixtures/tap.js --config ./tests/webpack.config.js',
    parser: tapParser,
    log: logger(logs)
  })

  await expect(brout.run()).resolves.toBe(0)
  expect(logs).toMatchSnapshot()
})

test('tap fail', async () => {
  const logs = []

  const brout = new Brout({
    url: 'http://127.0.0.1:3000',
    command: 'webpack serve ./tests/fixtures/tap-fail.js --config ./tests/webpack.config.js',
    parser: tapParser,
    log: logger(logs)
  })

  await expect(brout.run()).resolves.toBe(1)
  expect(logs).toMatchSnapshot()
})

test('uvu', async () => {
  const logs = []

  const brout = new Brout({
    url: 'http://127.0.0.1:3000',
    command: 'webpack serve ./tests/fixtures/uvu.js --config ./tests/webpack.config.js',
    parser: uvuParser,
    log: logger(logs)
  })

  await expect(brout.run()).resolves.toBe(0)
  expect(logs.filter(({ content }) => !content.includes('Duration'))).toMatchSnapshot()
})

test('uvu fail', async () => {
  const logs = []

  const brout = new Brout({
    url: 'http://127.0.0.1:3000',
    command: 'webpack serve ./tests/fixtures/uvu-fail.js --config ./tests/webpack.config.js',
    parser: uvuParser,
    log: logger(logs)
  })

  await expect(brout.run()).resolves.toBe(1)
  expect(logs.filter(({ content }) => !content.includes('Duration'))).toMatchSnapshot()
})
