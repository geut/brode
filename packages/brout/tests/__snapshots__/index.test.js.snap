// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
Array [
  Object {
    "content": "Target: chromium
",
    "type": undefined,
  },
  Object {
    "content": "log0",
    "type": "log",
  },
  Object {
    "content": "error0",
    "type": "error",
  },
  Object {
    "content": "warn0",
    "type": "warn",
  },
  Object {
    "content": "stdout0",
    "type": "stdout",
  },
  Object {
    "content": "stderr0",
    "type": "stderr",
  },
  Object {
    "content": "brout",
    "type": "log",
  },
]
`;

exports[`basic fail 1`] = `
Array [
  Object {
    "content": "Target: chromium
",
    "type": undefined,
  },
]
`;

exports[`tap 1`] = `
Array [
  Object {
    "content": "# target: chromium",
    "type": undefined,
  },
  Object {
    "content": "TAP version 13",
    "type": "log",
  },
  Object {
    "content": "# should result to the answer",
    "type": "log",
  },
  Object {
    "content": "ok 1 - answer should be 42",
    "type": "log",
  },
  Object {
    "content": "",
    "type": "log",
  },
  Object {
    "content": "1..1",
    "type": "log",
  },
  Object {
    "content": "# tests 1",
    "type": "log",
  },
  Object {
    "content": "# pass  1",
    "type": "log",
  },
  Object {
    "content": "# fail  0",
    "type": "log",
  },
  Object {
    "content": "# skip  0",
    "type": "log",
  },
]
`;

exports[`tap fail 1`] = `
Array [
  Object {
    "content": "# target: chromium",
    "type": undefined,
  },
  Object {
    "content": "TAP version 13",
    "type": "log",
  },
  Object {
    "content": "# should result to the answer",
    "type": "log",
  },
  Object {
    "content": "not ok 1 - answer should be 42",
    "type": "log",
  },
  Object {
    "content": "  ---",
    "type": "log",
  },
  Object {
    "content": "    actual: 42",
    "type": "log",
  },
  Object {
    "content": "    expected: 43",
    "type": "log",
  },
  Object {
    "content": "    operator: \\"equal\\"",
    "type": "log",
  },
  Object {
    "content": "    at: \\" eval (webpack-internal:///./tests/fixtures/tap-fail.js:7:5)\\"",
    "type": "log",
  },
  Object {
    "content": "  ...",
    "type": "log",
  },
  Object {
    "content": "",
    "type": "log",
  },
  Object {
    "content": "1..1",
    "type": "log",
  },
  Object {
    "content": "# tests 1",
    "type": "log",
  },
  Object {
    "content": "# pass  0",
    "type": "log",
  },
  Object {
    "content": "# fail  1",
    "type": "log",
  },
  Object {
    "content": "# skip  0",
    "type": "log",
  },
]
`;

exports[`uvu 1`] = `
Array [
  Object {
    "content": "[90mâ€¢ [39m",
    "type": "stdout",
  },
  Object {
    "content": "[32m  (1 / 1)
[39m",
    "type": "stdout",
  },
  Object {
    "content": "
  Total:     1",
    "type": "stdout",
  },
  Object {
    "content": "[32m
  Passed:    1[39m",
    "type": "stdout",
  },
  Object {
    "content": "
  Skipped:   0",
    "type": "stdout",
  },
  Object {
    "content": "
  Target:    chromium

",
    "type": "stdout",
  },
]
`;

exports[`uvu fail 1`] = `
Array [
  Object {
    "content": "[31mâœ˜ [39m",
    "type": "stdout",
  },
  Object {
    "content": "[31m  (0 / 1)
[39m",
    "type": "stdout",
  },
  Object {
    "content": "
  [1m[41m FAIL [22m[49m [2m\\"[22m[31m[1mMath.sqrt()[39m[22m[2m\\"[22m
    Expected values to be strictly equal:[3m[2m  (is)[23m[22m

        [32m++3    [2m[3m(Expected)[22m[23m[39m
        [31m--2    [2m[3m(Actual)[22m[23m[39m
        [90m
    at assert (webpack-internal:///../../node_modules/uvu/assert/index.mjs:48:8)
    at Module.is (webpack-internal:///../../node_modules/uvu/assert/index.mjs:56:2)
    at Object.eval [as handler] (webpack-internal:///./tests/fixtures/uvu-fail.js:8:43)
    at Number.runner (webpack-internal:///../../node_modules/uvu/dist/index.mjs:85:16)
    at exec (webpack-internal:///../../node_modules/uvu/dist/index.mjs:139:39)[39m


",
    "type": "stdout",
  },
  Object {
    "content": "
  Total:     1",
    "type": "stdout",
  },
  Object {
    "content": "[31m
  Passed:    0[39m",
    "type": "stdout",
  },
  Object {
    "content": "
  Skipped:   0",
    "type": "stdout",
  },
  Object {
    "content": "
  Target:    chromium

",
    "type": "stdout",
  },
]
`;
